cub/cub/util_type.cuh

```diff
@@ -980,50 +934,19 @@ struct CUB_DEPRECATED EnableIf
  * \brief Determine whether or not BinaryOp's functor is of the form <tt>bool operator()(const T& a, const T&b)</tt> or
  * <tt>bool operator()(const T& a, const T&b, unsigned int idx)</tt>
  */
+template <typename T, typename BinaryOp, typename = void>
+struct BinaryOpHasIdxParam : ::cuda::std::false_type
+{
+  CUB_DEPRECATED_BECAUSE("Use ::value instead") static constexpr bool HAS_PARAM = false;
```

Comment:
critical: our code (adjacent difference and discontinuity) is using `HAS_PARAM`. This change will lead to CUB code emitting deprecation warnings that users have no way of affecting. If you want to deprecate `HAS_PARAM`, make sure we do not rely on it.

Reply:
> Are users allowed/expected to specialize BinaryOpHasIdxParam?

Since this aspect is not documented, it's a gray area. I hope they don't 😄

Reply:
Alright, so adjacent difference and discontinuity will look for a nested `::value` from now on.